src/extra/gizmos/gizmos.cpp

```diff
@@ -83,6 +84,91 @@ struct TranslationGizmo : public Base {
     return outputTypes().elements[0];
   }
 };
-void registerGizmoShards() { REGISTER_SHARD("Gizmos.Translation", TranslationGizmo); }
+
+struct RotationGizmo : public Base {
+  static SHTypesInfo inputTypes() {
+    static Types inputTypes = {{CoreInfo::Float4x4Type, gfx::Types::Drawable}};
+    return inputTypes;
+  }
+  static SHTypesInfo outputTypes() { return CoreInfo::Float4x4Type; }
+  static SHOptionalString help() { return SHCCSTR("Shows a rotation gizmo "); }
+
+  // declare parameter named scale
+  PARAM_VAR(_scale, "Scale", "Gizmo scale", {CoreInfo::NoneType, CoreInfo::FloatType, CoreInfo::FloatVarType});
+  PARAM_IMPL(RotationGizmo, PARAM_IMPL_FOR(_scale));
+
+  // gizmo from translation_gizmo.hpp, not this file
+  gfx::gizmos::RotationGizmo _gizmo{};
+
+  SHVar activate(SHContext *shContext, const SHVar &input) {
+    float4x4 inputMat;
+    std::function<void(float4x4 & mat)> applyOutputMat;
```

Comment:
This might cause a very useless allocation depending on the standard.
I know it's part of the original implementation but even there should be fixed, pointers are not dangerous just use plain C constructs instead of if/else. KISS. cc @guusw

Reply:
I think this might not have a simple solution to it, but I may have some misunderstanding around it. Can @guusw confirm?

From what I understand, `applyOutputMat` has to be a function pointer rather than a plain C pointer because it is assigned a closure with captured variables, and it cannot be assigned to C pointers. There is a possible heavy cost due to virtual functions with std::function, but i'm not sure if the type erasure aspect of std::function is also whats allowing us to call `drawable->trs` and `drawable->transform`. I can't seem to declare `drawable` as a `Variant` in the closure provided to std::visit and still access those two members.